### PR TITLE
Post-merge cleanups on edit-user

### PR DESCRIPTION
**Pull request by jeremyzitomer-usds**
## Related Issue or Background Info

- Addresses PR feedback in https://github.com/CDCgov/prime-simplereport/pull/814

### CODE REVIEW

**Comment by jeremyzitomer-usds** [comment]
The comment threads with open questions from the last PR:

https://github.com/CDCgov/prime-simplereport/pull/814#discussion_r578467969 (about using the `User` (api model) class at the `ApiUserService` level to avoid pinging Okta twice)
https://github.com/CDCgov/prime-simplereport/pull/814#discussion_r578580295 (about making a change to changeset comments in `db.changelog-master.yaml`)
https://github.com/CDCgov/prime-simplereport/pull/814#discussion_r578777974 (about `EternalSystemManagedEntityRepository`)

Everything else either has corresponding changes in this PR or I've addressed in the previous PR.

**Comment by jeremyzitomer-usds** [comment]
Need to ask Skylight how to interpret the failing e2e tests tomorrow, it's unclear to me why that test is failing.

**Comment by jeremyzitomer-usds** [comment]
e2e issues fixed :) thanks @ClydeDroid and @nullflux !

File: frontend/e2e/nightwatch-entrypoint.bash [hunk 1]
```diff
@@ -5,7 +5,25 @@ function trapint {
     exit 0
 }
 
-echo 'Waiting for app to boot...'
+echo 'Waiting for backend to boot...'
```
Comment (jeremyzitomer-usds) [inline frontend/e2e/nightwatch-entrypoint.bash:8]: from Skylight, this should resolve race conditions between e2e tests and gradle bootRun :)

**Comment by jeremyzitomer-usds** [comment]
@benwarfield-usds just wanna make sure we're synced on the proper PR to review/sign off on, for edit-user changes

File: backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java [hunk 2]
```diff
@@ -133,44 +150,38 @@ private ApiUser getUser(UUID id) {
...
 
-    public Optional<ApiUser> getCurrentUserReadOnly() {
-        IdentityAttributes userIdentity = _supplier.get();
-        Optional<ApiUser> found = _apiUserRepo.findByLoginEmail(userIdentity.getUsername());
-        return found;
-    }
-
-    @Transactional(propagation = Propagation.REQUIRES_NEW)
-    public ApiUser getCurrentUser() {
+    public ApiUser getCurrentApiUser() {
```
Comment (benwarfield-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java:184]: This seems like it is only called internally—may as well make it private?

**Review by benwarfield-usds — COMMENTED** [review]
It looks fine to me, but I want to look it over one more time in the morning because I have a sense of having spent too much time looking at the trees and I think you added a couple substantively interesting forests that I should make sure I understand before hitting the button. Sending as a comment now since there's no reason to wait up for the few nitpicks and bits of cheerleading I've found to say so far.

**Comment by jeremyzitomer-usds** [comment]
@benwarfield-usds PR feedback from last night addressed, ready for "forest" review :)

File: frontend/e2e/nightwatch-entrypoint.bash [hunk 3]
```diff
@@ -5,7 +5,25 @@ function trapint {
...
+curl -k http://localhost:8080/health > /dev/null 2>&1
+result=$?
+
+polls=0
+while [[ $result -ne 0 && $polls -lt 120 ]]; do
```
Comment (benwarfield-usds) [inline frontend/e2e/nightwatch-entrypoint.bash:13]: Ten minutes is a long time to wait for startup of an app that only typically takes 15-30 seconds to start—I feel like this was originally a 2-minute wait and then the sleep got changed from 1 second to 5?

Comment (ClydeDroid) [reply frontend/e2e/nightwatch-entrypoint.bash:13]: Agreed, this definitely doesn't need to be as long

File: backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java [hunk 4]
```diff
@@ -135,46 +137,46 @@ public void updateUser(String oldUsername, IdentityAttributes userIdentity) {
...
         if (!users.iterator().hasNext()) {
             throw new IllegalGraphqlArgumentException("Cannot update role of Okta user with unrecognized username");
         }
         User user = users.single();
 
-        Set<String> orgExternalIds = new HashSet<>();
+        String orgId = org.getExternalId();
 
         // Remove user from old groups
         for (Group g : user.listGroups()) {
```
Comment (benwarfield-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java:154]: This nested loop was kind of confusing to read, though obviously the diffs didn't help. I think it is equivalent to the following, which I think is easier to follow:

    for (Group g : user.listGroups()) {
        if (g.getType() != GroupType.OKTA_GROUP) {
           continue;
        }
        String groupName = g.getProfile().getName();
        for (OrganizationRole r : OrganizationRole.values()) {
            if (r == OrganizationRole.USER) {
                continue;
            }
            if (generateGroupName(orgId, role).equals(groupName)) {
                g.removeUser(user.getId());
                break;
            }
        }
    }

If I'm right about that equivalence, then I think there's a further rearrangement that might be more efficient and possibly more readable:

    Set<String> roleGroups = Stream.of(OrganizationRole.values()).filter(r -> r != OrganizationRole.getDefault() && r != role)
            .map(r->generateGroupName(orgId, r))
            .collect(Collectors.toSet());
    for (Group g : user.listGroups()) {
        if (g.getType() == GroupType.OKTA_GROUP && roleGroups.contains(g.getProfile().getName()) {
           g.removeUser(user.getId());
        }
    }

File: backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java [hunk 5]
```diff
@@ -135,46 +137,46 @@ public void updateUser(String oldUsername, IdentityAttributes userIdentity) {
...
-                throw new IllegalGraphqlArgumentException("Cannot add Okta user to nonexistent group");
-            }
-            Group group = groups.single();
-            user.addToGroup(group.getId());
+        // Okta SDK's way of getting a group by group name
+        GroupList groups = _client.listGroups(groupName, null, null);
+        if (!groups.iterator().hasNext()) {
+            throw new IllegalGraphqlArgumentException("Cannot add Okta user to nonexistent group");
         }
+        Group group = groups.single();
```
Comment (benwarfield-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java:178]: Definitely a candidate for a factored-out method, if we ever have to look up individual groups again (which seems likely).

File: backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java [hunk 6]
```diff
@@ -135,46 +137,46 @@ public void updateUser(String oldUsername, IdentityAttributes userIdentity) {
...
+        user.addToGroup(group.getId());
 
         return role;
```
Comment (benwarfield-usds) [inline backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java:181]: Huh. That doesn't seem like a very useful return (in that it is one of the arguments to the method). Possible it should be boolean based on whether or not the person was already in the group? Or is there a way this is useful that I haven't noticed?

**Review by benwarfield-usds — APPROVED** [review]
Looks generally very good! Hard to follow some of the refactors without taking a long look on the big monitor, but I like them a lot.

A couple of further nitpicks and one hopefully-productive suggestion for streamlining some of the more involved bits of the Okta repo, and mostly some notes for future reference on directions we might want to take things as we go on. Thanks for getting this all done!

File: backend/src/test/java/gov/cdc/usds/simplereport/service/ApiUserServiceTest.java [hunk 7]
```diff
@@ -26,93 +26,48 @@ void setupData() {
...
     @Test
     @WithSimpleReportOrgAdminUser
     void getOrganizationRolesForUser_adminUser_success() {
         // This is the only way we can get a handle on the internal IDs of the users to check their roles
-        List<ApiUser> users = _service.getUsersInCurrentOrg(OrganizationRole.USER);
-        for (ApiUser user : users) {
-            Optional<OrganizationRoles> roles = _service.getOrganizationRolesForUser(user.getInternalId());
+        List<UserInfo> users = _service.getUsersInCurrentOrg();
+        for (UserInfo user : users) {
+            Optional<OrganizationRoles> roles = _service.getOrganizationRolesForUser(user.getId());
```
[the remark this answers is not in the document]
Comment (jeremyzitomer-usds) [reply backend/src/test/java/gov/cdc/usds/simplereport/service/ApiUserServiceTest.java:66]: i'll do one better -- i'm realizing that this test is the only code anywhere that calls `getOrganizationRolesForUser()`, so i'll delete the method and the test.

File: backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java (hunk 4 above)
Comment (jeremyzitomer-usds) [reply backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java:154]: good catch :)
